### ISSUE TITLE
Website function of OnionShare: Can a dynamic site be shared, e.g. like a websearch engine?

### ISSUE THREAD

**Issue by Sammysupport** [onionshare/onionshare#1910]
Hello,

I have two ideas and probably requests I would like to share with you, one here, the other probably later on..

I like Onionshare and also its four functions, I want to address a questions regarding the Website function:

Can this be extended to send not only a static webpage, but also a website, that is a websearch engine?
That might require a dynamic establishing of the website and also a query performance towards this website.

E.g. one user is creating a Websearch Engine about his collected RSS-Feeds.
That can be done e.g. over the Spot-On Application, see screenshots there and I post one below:   https://sammysupport.github.io/spot-on/   |  https://www.reddit.com/r/Spot_On_Encryption/

Can there be an **local_IP:Port_8080** be entered into a line of OnionShare and then the Websearch with such a page can be done over the website address of OnionShare?

Thanks for a comment and/or development in that regard.

What would be needed to have a visitor of the website beeing able to enter a search query term and get the dynamic results of that search engine back? Ideally in this case not html and png files are uploaded, but just one IP as string is provided: "192.168.178.15:8080" and the search works from any Tor Browser to the Spot-On Websearch function.

<img width="547" alt="Spot-On_Websearch_Webinterface" src="https://github.com/user-attachments/assets/147db5b2-6a3a-4706-8ae6-52df81ba45ac">

**Comment by Urban-Hacker** [comment]
Disclaimer, I’m not a dev of onion share but I think this cannot work, it’s way more complicated than a bunch of static html files being served and if you add enough dynamic capabilities to do this you would eventually re-invent a full server.

**Comment by SaptakS** [comment]
Agreed with what @Urban-Hacker said. Right now, OnionShare can only share static HTML since OnionShare is a tool used to share already available content, rather than implement a full server with backend capabilities. The website mode is almost similar to the share mode, with few added features so that the HTML actually gets rendered rather than showing up as a file in the browser.

**Comment by Sammysupport** [comment]
thanks for the first hints, @Urban-Hacker @SaptakS, yes that is the request.
Thought a kind of proxy or enhanced webrenderer can pick up the website-url to onion - probably it is called webserver.
I have an IP with port, or domain, or URL, and the string field for the URL or IP:Port in onionshare turns it into an onion address / website. The Websearch page with IP:Port as shown in the picture above should be shown onionized in Tor, so that users can enter a query term and get a new page back. 
Imagine a process like an IP-content hoster is referred to a domain: In that sense the IP and port should be referred to an onion address. Easy and simply, just entering the IP and Port in Onionshare.
Probably the App mentioned for the search has also a webserver Qt element implemented as here showing websites and search results is also possible. Now this resource is just forwarded in onionshare and covered addressable by an onion-string.
Something simple like that.
